toplevel/genprintval.ml

```diff
@@ -620,3 +620,22 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
     in nest tree_of_val max_depth obj (Ctype.correct_levels ty)
 
 end
+
+(* User-provided printers *)
+
+let printertypes_str = {|
+  type 'a printer_type_new = Format.formatter -> 'a -> unit
+  type 'a printer_type_old = 'a -> unit
+|}
+
+let printertypes_sig_ast =
+  Parse.interface (Lexing.from_string printertypes_str)
```

Comment:
Why not build the AST of the types directly using `Ast_helper`, instead of parsing the string representation?